spec/tasks/gen/action_spec.cr

```diff
@@ -3,31 +3,35 @@ require "../../spec_helper"
 include CleanupHelper
 
 describe Gen::Action do
-  it "generates a basic action" do
+  it "generates a basic browser action" do
     with_cleanup do
       io = IO::Memory.new
       valid_action_name = "Users::Index"
       ARGV.push(valid_action_name)
 
-      Gen::Action.new.call(io)
+      Gen::Action::Browser.new.call(io)
+
+      index_action = File.read("./src/actions/users/index.cr")
+      index_action.should contain(valid_action_name)
+      index_action.should contain("< BrowserAction")
```

Comment:
Since so much of these tests are duplicated, I think it would be helpful to extract a helper object for this. Here's what I'm thinking:

```crystal
# The `generate` method would generate the `io`, set the `ARGV` and call the generator and return the `io`
io = generate "Users::Index", with: Gen::Action::Browser 

# Custom expectation helper:
should_have_generated "#{valid_action_name} < BrowserAction", in: "./src/actions/users/index.cr"
```

Reply:
Yeah this is a good idea, it makes the tests clean and easy to read